File: uniapp_flutter/lib/models/gps_validation.dart

class GpsValidation {

  String _type;
  String _source;
  String _key;
  int _radius;
  String _keyType;

  GpsValidation.fromJson(Map<String, dynamic> parsedJson) {

    _type = parsedJson['type'];
    _source = parsedJson['source'];
    _key = parsedJson['key'];
    _radius = parsedJson['radius'];
    _keyType = parsedJson['key_type'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();

    if(this._type != null && this._type.isNotEmpty){
      data['type'] = this._type;
    }
    if(this._radius != null){
      data['radius'] = this._radius;
    }
    if(this._source != null && this._source.isNotEmpty){
      data['source'] = this._source;
    }
    if(this._key != null && this._key.isNotEmpty){
      data['key'] = this._key;
    }
    if(this._keyType != null && this._keyType.isNotEmpty){
      data['key_type'] = this._keyType;
    }

    return data;
  }

  String get type => _type;

  set type(String value) {
    _type = value;
  }

  String get source => _source;

  String get keyType => _keyType;

  set keyType(String value) {
    _keyType = value;
  }

  int get radius => _radius;

  set radius(int value) {
    _radius = value;
  }

  String get key => _key;

  set key(String value) {
    _key = value;
  }

  set source(String value) {
    _source = value;
  }


}

File: app/app/src/main/java/com/vassar/unifiedapp/model/GpsValidation.java
package com.vassar.unifiedapp.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonIgnoreProperties(ignoreUnknown = true)
public class GpsValidation implements Parcelable {

    public static final Creator<GpsValidation> CREATOR = new Creator<GpsValidation>() {
        @Override
        public GpsValidation createFromParcel(Parcel in) {
            return new GpsValidation(in);
        }

        @Override
        public GpsValidation[] newArray(int size) {
            return new GpsValidation[size];
        }
    };
    // bbox, circular
    @JsonProperty("type")
    private String type;
    // project, key
    @JsonProperty("source")
    private String source;
    @JsonProperty("key")
    private String key;
    @JsonProperty("radius")
    private double radius;
    @JsonProperty("key_type")
    private String keyType;

    public GpsValidation() {
    }

    private GpsValidation(Parcel in) {
        this.type = in.readString();
        this.source = in.readString();
        this.key = in.readString();
        this.radius = in.readDouble();
        this.keyType = in.readString();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.type);
        dest.writeString(this.source);
        dest.writeString(this.key);
        dest.writeDouble(this.radius);
        dest.writeString(this.keyType);
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getSource() {
        return source;
    }

    public void setSource(String source) {
        this.source = source;
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public double getRadius() {
        return radius;
    }

    public void setRadius(double radius) {
        this.radius = radius;
    }

    public String getKeyType() {
        return keyType;
    }

    public void setKeyType(String keyType) {
        this.keyType = keyType;
    }

    @Override
    public String toString() {
        return "GpsValidation{" +
                "type='" + type + '\'' +
                ", source='" + source + '\'' +
                ", key='" + key + '\'' +
                ", radius=" + radius +
                ", keyType='" + keyType + '\'' +
                '}';
    }
}
